src/functions-and-arrays.js

```diff
@@ -106,9 +209,65 @@ const matrix = [
   [1, 70, 54, 71, 83, 51, 54, 69, 16, 92, 33, 48, 61, 43, 52, 1, 89, 19, 67, 48]
 ];
 
-function greatestProduct() {}
+function greatestProduct(matrix) {
+  let biggest_hor_sum = 0;
+  let biggest_ver_sum = 0;
+
+  for (let array of matrix) {
+    for (let i = 0; i <= array.length - 4; i++) {
+      let current_sum = array[i] * array[i+1] * array[i+2] * array[i+3];
+      if (current_sum > biggest_hor_sum) {
+        biggest_hor_sum = current_sum;
+      }
+    }
+  }
+
+  for (let i = 0; i < matrix.length-1; i++) {
+    for (let z = 0; z <= matrix.length - 4; z++) {
+      let current_sum = matrix[z][i] * matrix[z+1][i] * matrix[z+2][i] * matrix[z+3][i]
+      if (current_sum > biggest_ver_sum) {
+        biggest_ver_sum = current_sum;
+      }
+    }
+  }
+
+  if (biggest_hor_sum > biggest_ver_sum) {
+    return biggest_hor_sum
+  } else {
+    return biggest_ver_sum
+  }
+}
+
 
 
+function greatestProductOfDiagonals(matrix) {
+  let biggest_product_left = 0;
+  let biggest_product_right = 0;
+
+  for (let i = 0; i <= matrix.length - 4; i++) {
+    for (let z = 0; z <= matrix.length -4; z++) {
+      let current_sum = matrix[i][z] * matrix[i+1][z+1] * matrix[i+2][z+2] * matrix[i+3][z+3];
```

Comment:
If you increment both `i` and `z`, you will "move" diagonal through the array. You would need to only increment `i` or `z` to "move" horizontally or vertically through the array.